src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs

```diff
@@ -420,32 +419,38 @@ public DirectoryInfo ComputeRootProjectBaseDir(IEnumerable<DirectoryInfo> projec
         /// Loads SARIF reports from the given projects and attempts to fix
         /// improper escaping from Roslyn V1 (VS 2015 RTM) where appropriate.
         /// </summary>
-        private void TryFixSarifReport(ProjectInfo project, string language, string reportFilePropertyKey)
+        private void TryFixSarifReport(ProjectInfo project, string language, string reportFilesPropertyKey)
         {
-            var tryResult = project.TryGetAnalysisSetting(reportFilePropertyKey, out Property reportPathProperty);
+            var tryResult = project.TryGetAnalysisSetting(reportFilesPropertyKey, out Property reportPathsProperty);
             if (tryResult)
             {
-                foreach (var reportPath in reportPathProperty.Value.Split(RoslynReportPathsDelimiter))
+                var newPropertyValueBuilder = new StringBuilder();
+                project.AnalysisSettings.Remove(reportPathsProperty);
+                foreach (var reportPath in reportPathsProperty.Value.Split(RoslynReportPathsDelimiter))
                 {
                     var fixedPath = fixer.LoadAndFixFile(reportPath, language);
 
-                    if (!reportPath.Equals(fixedPath)) // only need to alter the property if there was no change
+                    if (!reportPath.Equals(fixedPath)) // only need to alter the property if there was a change
                     {
-                        // remove the property ahead of changing it
-                        // if the new path is null, the file was unfixable and we should leave the property out
-                        project.AnalysisSettings.Remove(reportPathProperty);
-
                         if (fixedPath != null)
                         {
-                            // otherwise, set the property value (results in no change if the file was already valid)
-                            var newReportPathProperty = new Property
-                            {
-                                Id = reportFilePropertyKey,
-                                Value = fixedPath,
-                            };
-                            project.AnalysisSettings.Add(newReportPathProperty);
+                            newPropertyValueBuilder.Append($"{fixedPath}{RoslynReportPathsDelimiter}");
                         }
                     }
+                    else
+                    {
+                        newPropertyValueBuilder.Append($"{reportPath}{RoslynReportPathsDelimiter}");
+                    }
+                }
+
+                if (newPropertyValueBuilder.Length > 0)
+                {
+                    var newReportPathProperty = new Property
+                    {
+                        Id = reportFilesPropertyKey,
+                        Value = newPropertyValueBuilder.ToString().TrimEnd(RoslynReportPathsDelimiter)
```

Comment:
I personally don't like build & trim approach. In VB.NET, I'd do it in single-liner
```VB.NET
  If newPropertyValueBuilder.Length <> 0 Then newPropertyValueBuilder.Append(RoslynReportPathsDelimiter)
```
In C#, it's not so nice. Can you build a list and do a `string.Join` instead?

Reply:
Sure.